### PR TITLE
chore(build): upgrade to Babel 7

### PR DESCRIPTION
**Pull request by asudoh**
Upgrade to Babel7 allows better tree-shaking for ES classes, though we need an extra step for class properties (done by an additional Babel plugin).

Better tree-shaking helps ensure `carbon-components-react` bundle only include `settings.js` in this library.

Refs IBM/carbon-components-react#1620.

#### Changelog

**New**

- A Babel plugin to ensure class properties can be tree-shaked.

**Changed**

- Upgrade to Babel7.
- Stop using spread operator (the Babel helper for array spread operator creates dependency to `Symbol`, breaking IE11)

#### Testing / Reviewing

Testing should make sure no component/build is broken.

### CODE REVIEW

File: src/components/accordion/accordion.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ class Accordion extends mixin(createComponent, initComponentBySearch, handles) {
    * The map associating DOM element and accordion UI instance.
    * @type {WeakMap}
    */
-  static components = new WeakMap();
+  static components /* #__PURE__ */ = new WeakMap();
```
Comment (joshblack) [inline src/components/accordion/accordion.js]: Are we safe doing this versus having a compiler/build tool like webpack insert these?

File: tools/rollup.config.dev.js [hunk 2]
```diff
@@ -48,11 +48,9 @@ module.exports = {
     }),
     babel({
       exclude: ['node_modules/**'],
-      plugins: ['external-helpers'],
```
Comment (joshblack) [inline tools/rollup.config.dev.js:51]: Any reason for dropping this plugin?

**Review by joshblack — COMMENTED** [review]
Left a couple of comments. Looks great! 🎉

File: tools/rollup.config.dev.js (hunk 2 above)
Comment (asudoh) [reply tools/rollup.config.dev.js:51]: @joshblack https://github.com/rollup/rollup-plugin-babel#helpers is the best resource. In short, Rollup requires using `babel-plugin-external-helpers` with Babel 6, but not the case for Babel 7.

File: src/components/accordion/accordion.js (hunk 1 above)
Comment (asudoh) [reply src/components/accordion/accordion.js]: @joshblack Good question, I made a change to rename the hint to avoid the conflict with `/* #__PURE__ */`.

**Review by aledavila — APPROVED** [review]
LGTM

**Review by joshblack — APPROVED** [review]

**Pull request merged**